crates/proc-macro-srv/src/lib.rs

```diff
@@ -66,13 +68,18 @@ impl ProcMacroSrv {
         // FIXME: replace this with std's scoped threads once they stabilize
         // (then remove dependency on crossbeam)
         let result = crossbeam::scope(|s| {
-            let res = s
+            let res = match s
+                .builder()
+                .stack_size(EXPANDER_STACK_SIZE)
+                .name(task.macro_name.clone())
                 .spawn(|_| {
                     expander
                         .expand(&task.macro_name, &macro_body, attributes.as_ref())
                         .map(|it| FlatTree::new(&it))
-                })
-                .join();
+                }) {
+                Ok(handle) => handle.join(),
+                Err(e) => std::panic::resume_unwind(Box::new(e)),
```

Comment:
I might be missing something, but wouldn't `unwrap` be fine here?

Sorry for all the nitpicks.

Reply:
https://github.com/rust-lang/rust-analyzer/pull/12835#discussion_r926963191

Reply:
Does that still apply if this is a simple `io::Error`? I'd have to test.

Reply:
Well, whatever :laughing:.

@bors r+